test/checks/navigation/identical-links-same-purpose.js

```diff
@@ -0,0 +1,137 @@
+describe('identical-links-same-purpose tests', function() {
+	'use strict';
+
+	var fixture = document.getElementById('fixture');
+	var queryFixture = axe.testUtils.queryFixture;
+	var check = checks['identical-links-same-purpose'];
+	var checkContext = axe.testUtils.MockCheckContext();
+	var options = {};
+
+	afterEach(function() {
+		fixture.innerHTML = '';
+		checkContext.reset();
+		axe._tree = undefined;
+	});
+
+	it('returns undefined for native link with `href` but no accessible name', function() {
+		var vNode = queryFixture('<a id="target" href="/home/#/foo"></a>');
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isUndefined(actual);
+		assert.isNull(checkContext._data);
+	});
+
+	it('returns undefined when ARIA link that has no accessible name', function() {
+		var vNode = queryFixture('<span role="link" id="target"></span>');
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isUndefined(actual);
+		assert.isNull(checkContext._data);
+	});
+
+	it('returns undefined when ARIA link has only any combination of unicode (emoji, punctuations, nonBmp) characters as accessible name', function() {
+		var vNode = queryFixture(
+			'<button id="target" role="link">☀️!!!₨   </button>'
+		);
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isUndefined(actual);
+		assert.isNull(checkContext._data);
+	});
+
+	it('returns true for native links with `href` and accessible name', function() {
+		var vNode = queryFixture('<a id="target" href="/home/#/foo">Pass 1</a>');
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isTrue(actual);
+		assert.hasAllKeys(checkContext._data, ['name', 'urlProps']);
+		assert.equal(checkContext._data.name, 'Pass 1'.toLowerCase());
+		assert.equal(checkContext._data.urlProps.hash, '#/foo');
+		assert.equal(checkContext._data.urlProps.pathname, '/home');
+	});
+
+	it('returns undefined for `AREA with closest `MAP` with no name attribute', function() {
```

Comment:
These AREA things are done by the matcher. These tests don't really do anything.